psalm.xml

```diff
@@ -59,6 +59,10 @@
         <RedundantCondition errorLevel="suppress"/>
         <RedundantCast errorLevel="suppress"/>
         <RedundantCastGivenDocblockType errorLevel="suppress"/>
+
+        <!-- Not using the result of pure functions is common within PSL -->
+        <!-- e.g: with and Psl\invariant_violations() -->
+        <UnusedFunctionCall errorLevel="suppress" />
```

Comment:
not sure if this should be considered a bug in psalm or not.

1. pure function are considered to have no side effects
2. calling a function with no side effects without using it's results is useless
3. psalm consider throwing an exception not a side effect
4. hacklang considers throwing an exception a side effect ( https://github.com/facebook/hhvm/blob/master/hphp/hack/doc/HIPs/contexts_and_coeffects.md )
5. if throwing an exception is a side effect, this is most likey to result in all functions being non pure, as 90% of calls result in an exception
6. if throwing an exception is not a side effect, how can a pure function tell psalm that it's okay not to use it's result?

wdyt @muglug?
 
/cc @ocramius